### PR TITLE
SGF sorting

### PR DESCRIPTION
**Pull request by bhostetler18**
This PR starts to address #2013 by adding various sort options and an ascending/descending toggle. It's not the prettiest, but the underlying functionality is mostly there. Currently, the three sort modes are `name`, `date added`*, and `game date`. Other suggestions welcome, especially if anyone has ideas for a better and more consistent UI.

*For this sort mode I currently use the `created` field on each game, but I'm not sure if that's appropriate. If that field doesn't represent the date a game was uploaded, would `entry_id` or `game_id` work instead (I'm assuming one of those is a sequential, increasing value)? That might be better even if `created` behaves as expected, since it would avoid parsing a new `Date` during each sort. @anoek 

~~In order to test this functionality, I've just been manually adding fake game entries since the beta server doesn't appear to save anything (as mentioned in #2020). Still curious if there's a better way.~~
Fixed with #2027 

sort

### CODE REVIEW

**Comment by bhostetler18** [comment]
sorting

I've made a layout similar to the game list page. It currently defaults to sorting the most recently added files on top since that seemed popular. Let me know if anything seems amiss or you want something else with the layout/functionality!

File: src/views/LibraryPlayer/LibraryPlayer.tsx [hunk 1]
```diff
@@ -69,9 +74,27 @@ class _LibraryPlayer extends React.PureComponent<LibraryPlayerProperties, Librar
             games_checked: {},
             new_collection_name: "",
             new_collection_private: false,
+            sort_order: "date_added",
+            sort_descending: true,
```
Comment (anoek) [inline src/views/LibraryPlayer/LibraryPlayer.tsx:78]: I reckon it'd be nice if we remembered these, that's usually done with the preferences system, which goes something like this

1) Add your preference keys and default values to `lib/preferences.ts", so something like `"sgf.sort-order": "date_added"`
2) `import * as  preferences from "preferences";`
3) 
```
sort_order: preferences.get("sgf.sort-order"),
sort_descending: preferences.get("sgf.sort-descending"),
```
4) When you change your sort order, call `preferences.set("sgf.sort-order", whatever)`

Comment (bhostetler18) [reply src/views/LibraryPlayer/LibraryPlayer.tsx:78]: Will do!

**Comment by anoek** [comment]
Awesome, thanks for the patch and going the extra mile!

**Pull request merged**